pkg/kubectl/cmd/cp/cp.go

```diff
@@ -275,11 +275,42 @@ func (o *CopyOptions) copyToPod(src, dest fileSpec, options *exec.ExecOptions) e
 	return o.execute(options)
 }
 
+// Changes the destination filename and location if they are not explicitly specified.
+// If destination is a directory, filename will be the same as the source.
+// Also, symlink destinations are converted to true paths
+func convertHostDestFilename(srcFile string, destFile string) (string, error) {
+	stat, err := os.Stat(destFile)
```

Comment:
We only need to `EvalSymlinks` for destFile here. No more steps are needed.

Reply:
The symlink steps were to address the other bug that my PR addresses, which is `kubectl cp` failing to symlinks on full paths. Keeping those bugs I address in mind should make more clear why the code was created.